// GraphView/GraphChart.js

import React, { Component } from 'react';

import EmptyGraph      from './EmptyGraph.js'
import SingleItemGraph from './SingleItemGraph.js'
import LineGraph       from './LineGraph.js'


class GraphChart extends Component {

  constructor(props) {
    super(props);

    // this.graphName = this.props.currentGraphObj.key;

    this.state = {
      currentName: this.props.currentGraphObj.key,
      currentGraph: this.props.currentGraphObj.value
    };

    console.log('GraphChart props', props);
  }

  graphDataLength(graphData) {
    if (!graphData.length) {
      return 0;
    }
    return graphData.length;
  };

  selectGraphElement(state) {
    const graphName = this.state.currentName;
    const graphData = this.state.currentGraph;
    const len = this.graphDataLength(graphData);

    if (len === 0) return <EmptyGraph/>;
    if (len === 1) return <SingleItemGraph graphName={graphName} itemKey={graphData[0].number} itemValue={graphData[0].datetime} />;
    return <LineGraph graphName={graphName} graphData={graphData} />
  };

  render() {
    return (
        <div className="GraphChart">
          <h3>{this.state.currentName}</h3>

          {this.selectGraphElement(this.state)}
        </div>
    );
  }
}

export default GraphChart;
