// GraphView/GraphInput.js

import React, { Component } from 'react';

class GraphInput extends Component {

  constructor(props) {
    super(props);
    console.log('Graph Input props', props);
    this.updateGraphNow = props.updateGraphNow;
    this.state = {
      numberInput: '',
      nameInput: this.props.currentGraphObj.key
    };

    this.handleNumberInputChange = this.handleNumberInputChange.bind(this);
    this.handleNameInputChange = this.handleNameInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleNumberInputChange(event) {
    console.log("Number Input event.target", event.target);
    this.setState({numberInput: event.target.value});
  }

  handleNameInputChange(event) {
    console.log("Name Input event.target", event.target);
    this.setState({nameInput: event.target.value});
  }

  handleSubmit(event) {
    console.log('Form Submitted (name, number): ', this.state.nameInput, this.state.numberInput);
    if(this.state.nameInput.length){
      console.log('Submitting', this.state.nameInput, this.state.numberInput);
      this.updateGraphNow(this.state.nameInput, this.state.numberInput);
    }
    // this.updateGraph(this.state.numberInput);
    event.preventDefault();
  }

  render() {
    return (
        <div className="GraphInput">
          <h3>Graph It</h3>
          <div className="GraphInput-name">
            <h5>{this.props.currentGraphObj.key}</h5>
          </div>
          <form onSubmit={this.handleSubmit}>
            <label>
              Name:
              <input type="text" value={this.state.nameInput} onChange={this.handleNameInputChange} />
            </label>
            <label>
              Number:
              <input type="number" value={this.state.numberInput} onChange={this.handleNumberInputChange} />
            </label>
            <input type="submit" value="Save" />
          </form>

        </div>
    );
  }
}

export default GraphInput;
