// GraphView/GraphChart.js

import React, { Component } from 'react';

import EmptyGraph      from './EmptyGraph.js'
import SingleItemGraph from './SingleItemGraph.js'
import LineGraph       from './LineGraph.js'


class GraphChart extends Component {

  constructor(props) {
    super(props);

    // this.graphName = this.props.currentGraphObj.key;
    // this.graphData = this.props.currentGraphObj.value;

    this.state = {
      currentName: this.props.currentGraphObj.key,
      currentGraph: this.props.currentGraphObj.value
    };

    console.log('GraphChart props', props);
  }

  graphDataLength(graphData) {
    if (!graphData.length) {
      return 0;
    }
    return graphData.length;
  };

  selectGraphElement(graphName, graphData) {
    // const graphName = this.state.currentName;
    // const graphData = this.state.currentGraph;
    const len = this.graphDataLength(graphData);

    if (len === 0) return <EmptyGraph/>;
    if (len === 1) return <SingleItemGraph graphName={graphName} itemKey={graphData[0].number} itemValue={graphData[0].datetime} />;
    return <LineGraph graphName={graphName} graphData={graphData} />
  };

  // componentDidMount() {
  //   this.selectGraphElement(this.state);
  // }

  render() {
    console.log('Rendering Graph Chart', this.props, this.state);
    const graphName = this.props.currentGraphObj.key;
    const graphData = this.props.currentGraphObj.value;
    return (
        <div className="GraphChart">
          <h3>{graphName}</h3>

          {this.selectGraphElement(graphName, graphData)}
        </div>
    );
  }
}

export default GraphChart;
