// GraphView/GraphList.js

import React, { Component } from 'react';

// maybe break out to it's own file at some point
const GraphLink = (props) =>
    <li>
      <span  onClick={props.handleClick.bind(null, props.graphName)}>{props.graphName}</span>
    </li>;


class GraphList extends Component {

  constructor(props) {
    super(props);
    console.log('Graph List props', props);
    // this.persistFn = props.persistGraphObj;
    this.state = {allGraphs: props.allGraphs};

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(graphName) {
    console.log('GraphList click:', graphName);
    this.props.displayGraph(graphName);
    event.preventDefault();
    
    
  }

  graphList() {
    const graphList = Object.keys(this.state.allGraphs).map((graphName, index) => {
      let idxString = "graphList-" + index;
      return (
        <GraphLink key={idxString} handleClick={this.handleClick} graphName={graphName} />
      )});
    return graphList;
  };
  
  render() {
    return (
        <div className="GraphList">
          <h3>Graph List</h3>
          {this.graphList()}
        </div>
    );
  }
}


export default GraphList;
