// GraphView/GraphChart.js

import React, { Component } from 'react';

import EmptyGraph      from './EmptyGraph.js'
import SingleItemGraph from './SingleItemGraph.js'
import LineGraph       from './LineGraph.js'


class GraphChart extends Component {

  constructor(props) {
    super(props);

    this.props.currentGraphObj.value.push({"number": -1, "datetime": "2016-12-10T15:21:25.938Z"});
    // this.props.currentGraphObj.value.push({"number": 0, "datetime": "2016-12-11T15:21:26.938Z"});

    this.state = {
      currentName: this.props.currentGraphObj.key,
      currentGraph: this.props.currentGraphObj.value
    };

    console.log('GraphChart props', props);

    // let currentGraphObj = this.state.currentGraph;
    // let name = Object.keys(currentGraphObj)[0];
    //
    // // This is temporary ... it's just to put some data into the storage
    // let graphData = currentGraphObj[name];
    // graphData[name].push({"number": -1, "datetime": "2016-12-10T15:21:25.938Z"});
    // graphData[name].push({"number": 0, "datetime": "2016-12-11T15:21:26.938Z"});
    //
    
    // this.state = { currentGraphObj };
    //   graphData: [
    //     {"number": -1, "datetime": "2016-12-10T15:21:25.938Z"},
    //     {"number": 0, "datetime": "2016-12-11T15:21:26.938Z"},
    //     {"number": 1, "datetime": "2016-12-12T15:21:27.938Z"},
    //     {"number": -2, "datetime": "2016-12-13T15:21:33.014Z"},
    //     {"number": 3, "datetime": "2016-12-14T15:21:34.014Z"},
    //     {"number": 4, "datetime": "2016-12-15T15:21:36.650Z"},
    //     {"number": 8, "datetime": "2016-12-23T15:21:37.650Z"},
    //     {"number": 4, "datetime": "2016-12-25T12:25:39.650Z"}
    //   ]
    // };
    // this.state = {
    //   graphData: []
    // };
    // this.state = {
    //   graphData: [{"number": 1, "datetime": "2016-12-10T15:21:27.938Z"}]
    // }
  }

  // updateGraphData(key, newData) {
  //   this.state
  // }
  
  currentGraphName(state) {
    
  }

  graphDataLength(graphData) {
    if (!graphData.length) {
      return 0;
    }
    return graphData.length;
  };

  selectGraphElement(state) {
    //TODO: Our state has te be maintained at a higher level than here
    // console.log('GraphChart currentGraph', state.currentGraph);
    const graphName = this.state.currentName;
    const graphData = this.state.currentGraph;
    const len = this.graphDataLength(graphData);
    // console.log('GraphChart graphData', graphData);

    if (len === 0) return <EmptyGraph/>;
    if (len === 1) return <SingleItemGraph graphName={graphName} itemKey={graphData[0].number} itemValue={graphData[0].datetime} />;
    return <LineGraph graphName={graphName} graphData={graphData} />
  };

  render() {
    return (
        <div className="GraphChart">
          <h3>Graph Chart</h3>

          {this.selectGraphElement(this.state)}
        </div>
    );
  }
}

export default GraphChart;
