// GraphView/GraphInput.js

import React, { Component } from 'react';

class GraphInput extends Component {

  constructor(props) {
    super(props);
    console.log('Graph Input props', props);
    this.updateGraphNow = props.updateGraphNow;
    this.state = {
      numberInput: '',
      nameInput: this.graph(this.props.currentGraphObj).name
    };

    this.handleNumberInputChange = this.handleNumberInputChange.bind(this);
    this.handleNameInputChange = this.handleNameInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  isNumeric(val) {
    // must be == (not ===)
    return Number(parseFloat(val))==val; //eslint-disable-line eqeqeq
  }

  handleNumberInputChange(event) {
    this.setState({numberInput: event.target.value});
  }

  handleNameInputChange(event) {
    this.setState({nameInput: event.target.value});
  }

  handleSubmit(event) {
    // TODO: Show error if no number was entered rather than ignoring
    if (this.isNumeric(this.state.numberInput)) {
      if(this.state.nameInput.length){
        this.updateGraphNow(this.state.nameInput, this.state.numberInput);
      }
    }
    event.preventDefault();
    this.setState({numberInput: ''});
  }

  graph(graphObj) {
    console.log('graph', graphObj);
    const graphObjKeys = Object.keys(graphObj);
    console.log('graph keys', graphObjKeys);
    const graphObjSize = graphObjKeys.length;
    if (graphObjSize !==1) {
      console.error('Expected a single graph object, but got ', graphObjSize);
    }
    const graphName = graphObjKeys[0];
    const graphData = graphObj[graphName];
    console.log('graph returning', graphName, graphData);
    return { name: graphName, data: graphData };
  }

  render() {
    return (
        <div className="GraphInput">
          <h3>Graph It</h3>
          <div className="GraphInput-name">
            <h5>{this.graph(this.props.currentGraphObj).name}</h5>
          </div>
          <form onSubmit={this.handleSubmit}>
            <label>
              Name:
              <input type="text" value={this.state.nameInput} onChange={this.handleNameInputChange} />
            </label>
            <label>
              Number:
              <input type="number" value={this.state.numberInput} onChange={this.handleNumberInputChange} />
            </label>
            <input type="submit" value="Save" />
          </form>
        </div>
    );
  }
}

export default GraphInput;
