// StorageModel/Storage.js

import History from './History';

// ToDo Needs testing

class LocalStorageWrapper {
  constructor(mainKey) {
    this.mainKey = '__justgraphit__';
  }

  load = () => {
    return JSON.parse(localStorage.getItem(this.mainKey));
  };

  save = (value) =>  {
    localStorage.setItem(this.mainKey, JSON.stringify(value));
  };
}

class Storage {

  constructor() {
    this.persistentStore = new LocalStorageWrapper();
    this.store = this.persistentStore.load();
    if (this.isEmpty()) {
      this.persistentStore.save({});
    }
    this.history = new History(2);
  }

  deleteObject(key) {
    const deleteRetVal = (delete this.store[key]);
    if (deleteRetVal) {
      this.updatePersistence();
    }
    return deleteRetVal;
  }
  
  getAll() { return this.store; };

  getLabeledObject(storageObj) {
    const storageObjKeys = Object.keys(storageObj);
    const storageObjSize = storageObjKeys.length;
    if (storageObjSize !==1) {
      console.error('Expected a single stored object, but got ', storageObjSize);
    }
    const objName = storageObjKeys[0];
    const objData = storageObj[objName];
    return { name: objName, data: objData };
  }
  
  getLastKey() {
    return this.history.getLast();
  }

  // returns last object from history
  getLastStorageObject() {
    let obj = {};
    const key = this.getLastKey();
    obj[key] = this.load(key);
    return obj;
  };
  
  isEmpty= () => {
    if (!this.store || !this.store.length) {
      return true;
    }
    return Object.keys(this.store).length === 0;
  };

  load(key) {
    return this.store[key];
  };

  save(key, val) {
    this.updateHistory(key);
    if (this.store === null) {
      this.store = {};
    }
    this.store[key] = val;
    this.updatePersistence();
    return key;
  };

  updateHistory(key) {
    if (key && key.length) {
      this.history.push(key);
    }
  }

  updatePersistence = () => {
    this.persistentStore.save(this.store);
  }
}

export default Storage;
