// components/GraphChooser/GraphChooser.js

import React, { Component } from 'react';

import EmptyGraph      from 'components/GraphTypes/EmptyGraph.js'
import SingleItemGraph from 'components/GraphTypes/SingleItemGraph.js'
import LineGraph       from 'components/GraphTypes/LineGraph.js'

class GraphChooser extends Component {

  constructor(props) {
    super(props);
    //this.handleChange = this.handleChange.bind(this);
    //this.handleSignUp = this.handleSignUp.bind(this);

    this.state = {
      graphData: [
        {"number": 1, "datetime": "2016-12-10T15:21:27.938Z"},
        {"number": 2, "datetime": "2016-12-10T15:21:33.014Z"},
        {"number": 3, "datetime": "2016-12-10T15:21:37.650Z"}
      ]
    };
    // this.state = {
    //   graphData: []
    // };
    // this.state = {
    //   graphData: [{"number": 1, "datetime": "2016-12-10T15:21:27.938Z"}]
    // }
  }

  graphDataLength(graphData) {
    if (!graphData.length) {
      return 0;
    }
    return graphData.length;
  };

  selectGraphElement(state) {
    const graphData = state.graphData;
    const len = this.graphDataLength(graphData);

    if (len === 0) return <EmptyGraph/>;
    if (len === 1) return <SingleItemGraph itemKey={graphData[0].number} itemValue={graphData[0].datetime} />;
    return <LineGraph graphData={graphData} />
  };

  render() {
    return (
        <div className="GraphInput">
          <h3>Graph Chart</h3>

          {this.selectGraphElement(this.state)}
        </div>
    );
  }
}

export default GraphChooser;
