// EditorView/Edit.js

import React, { Component } from 'react';
import {cyan500, cyan100} from 'material-ui/styles/colors';
import ActionDelete from 'material-ui/svg-icons/action/delete';


import {Table, TableBody, TableHeader, TableHeaderColumn, TableRow, TableRowColumn} from 'material-ui/Table';

class Edit extends Component {

  constructor(props) {
    super(props);
    console.log('Edit props', props);
    this.deleteItem = this.deleteItem.bind(this);
    this.getLabeledObject = props.storage.getLabeledObject;
    this.state = {};
    this.state.graphData = this.getGraphData();
  }

  deleteItem = (index) => {
    const graphName = this.getGraphName();
    let graphData = this.getGraphData();
    const item = graphData[index];
    const itemStr = "value: " + item.number + ", date: " + item.datetime;
    if(confirm('Delete item: ' + itemStr)) {
      if (graphData.splice(index, 1)) {
        this.props.storage.save(graphName, graphData);
      }
      this.setState({graphData: this.getGraphData()});
      console.log('Deleted:', graphData[index]);
    }
  };

  getGraphName = () => {
    // TODO: Validate just one exists
    return Object.keys(this.graphObj())[0];
  };

  getGraphData = () => {
    return this.graphObj()[this.getGraphName()];
  };

  graphObj = () => {
    return this.props.storage.getLastStorageObject() || {};
  };

  editListItems() {
    const storageObj = this.graphObj(); 
    console.log('storageObj', storageObj);
    console.log('getGraphName', this.getGraphName());
    console.log('getGraphData', this.getGraphData());
    if (!storageObj) {
      return '';
    }
    // const editList = Object.keys(storageObj).map((dataKey, thing) => {
    //   let dataList = storageObj[dataKey];
    //   console.log('dataList', dataList, thing);
    //   return dataList.map((dataItem, index) => {
    //     let idxString = "editList-" + index;
    //     console.log('dataItem', dataItem);
    //     return (
    //         // <ListItem primaryText={dataItem.datetime} rightIcon={<ActionInfo />} />
    //         <TableRow>
    //           <TableRowColumn>{dataItem.datetime}</TableRowColumn>
    //           <TableRowColumn>{dataItem.number}</TableRowColumn>
    //           <TableRowColumn>
    //             <ActionDelete color={cyan500} hoverColor={cyan100} onClick={() => this.deleteItem(index)}/>
    //           </TableRowColumn>
    //         </TableRow>
    //     );
    //   });
    // });
    const editList = this.state.graphData.map((dataItem, index) => {
        let idxString = "editList-" + index;
        console.log('dataItem', dataItem);
        return (
            // <ListItem primaryText={dataItem.datetime} rightIcon={<ActionInfo />} />
            <TableRow key={idxString}>
              <TableRowColumn>{dataItem.datetime}</TableRowColumn>
              <TableRowColumn>{dataItem.number}</TableRowColumn>
              <TableRowColumn>
                <ActionDelete color={cyan500} hoverColor={cyan100} onClick={() => this.deleteItem(index)}/>
              </TableRowColumn>
            </TableRow>
        );
      });


    return editList;
  };

  tableWrapper() {
    return (
        <Table>
          <TableHeader displaySelectAll={false}>
            <TableRow>
              <TableHeaderColumn>Datetime</TableHeaderColumn>
              <TableHeaderColumn>Number</TableHeaderColumn>
              <TableHeaderColumn>Action</TableHeaderColumn>
            </TableRow>
          </TableHeader>
          <TableBody displayRowCheckbox={false}>
            {this.editListItems()}
          </TableBody>
        </Table>
    );
  }

  render() {
    return (
      <div className="EditView">
        <h3>{this.getLabeledObject(this.graphObj()).name}</h3>
          {this.tableWrapper()}
        </div>
    )
  }
}

export default Edit;
